Packages/com.unity.inputsystem/InputSystem/Plugins/UI/InputSystemUIInputModule.cs

```diff
@@ -1615,7 +1615,10 @@ private bool CheckForRemovedDevice(ref InputAction.CallbackContext context)
             return false;
         }
 
-        internal const float kPixelPerLine = 20;
+        // Due to different input systems at Unity all piping input into UI,
+        // we need to have a consistent scroll speed across the board.
+        // Hence this constant should be equal to Windows WHEEL_DELTA constant.
+        internal const float kPixelPerLine = 120.0f;
```

Comment:
This sounds Windows-specific. What's the effect on Mac? Will it scroll at super high speed there now? Should we #if this?

Reply:
Just tried on my macbook 2017 model, printing `Mouse.current.scroll` values:
- Touch pad provides values in range 0-100 on average, starting from 5-10 and going to 50-100 when I scroll hard.
- Mouse wheel starts with 1-2 for scrolling single wheel notches, but then goes to 50-110 when I scroll faster.

So macOS implements acceleration curve there ... meaning oh well .. that `kPixelPerLine` is completely broken as well?
Maybe whole concept of "scrolling in lines" doesn't make sense, and it should be "scrolling in pixels", meaning `StandaloneInputModule` should use pixel based scrolling and `kPixelPerLine` be removed?

Seems like this is a much bigger change than just our package

Reply:
Agree that "scrolling in lines" isn't a good concept. Think `InputSystemUIInputModule` is the only place where we should attempt the thing. Basically, there the guiding thing is that swapping out `StandaloneInputModule` should (more or less) not result in any visible change in behavior. So whatever the old thing ends up doing for scrolling, we'll need to try to emulate.

Given that the old thing does the scroll amount adjustment already at the platform level, probably means we'll have to actually handle the `kPixelPerLine` thing in some platform-dependent way here instead of just one simple constant.

IMO we don't need to be 100%. If the scrolling behavior is reasonably close, IMO that's good enough.

Reply:
@Rene-Damm we can do `clamp((int)scroll.x, -1, 1)` and that would be almost 100% corresponding to `Input.mouseScrollDelta`. But it's insanely broken, it means that your scroll speed is FPS + mouse event rate dependent.

Reply:
Also on Windows, when I scroll my mouse wheel really fast, I can sometimes get line values larger than 1 from `Input` (the `Input.mouseScrollDelta` value can return (0, -2) for instance).

Reply:
Hm, I assume that is because it's clamping on event .. so if there are two events in the frame .. you get 2, oh well why it's so hard 🤣